### PR TITLE
VueJS Test

### PR DESCRIPTION
**Pull request by eclimov**
**Points done**
- Paginated list of issues
- Filter by status: closed / opened / all
- Page to see the info about specific issue
- Integrate design component framework: vuetify
- Integrate VueRouter
- Integrate Vuex

**Additional done**
- Loader while data is fetched
- Configurable number of issues per page
- Save page number after reload
- Preview user information on hover avatar

### CODE REVIEW

File: src/components/issues/List.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          :length="paginationPagesCount"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters } from 'vuex'
+import Progress from '../Progress'
```
Comment (rapi) [inline src/components/issues/List.vue:52]: dont use ..

File: src/components/layouts/LeftSidebar.vue [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import StatusFilter from '@/components/leftSidebar/StatusFilter'
+
+export default {
+  name: 'LeftSidebar',
+  components: { StatusFilter }
+}
+</script>
+
+<style scoped>
```
Comment (rapi) [inline src/components/layouts/LeftSidebar.vue:17]: Dont leave empty styles

File: src/components/user/Preview.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  props: {
+    user: {
+      type: Object,
+      required: true
+    }
+  }
+}
+</script>
+
+<style scoped>
```
Comment (rapi) [inline src/components/user/Preview.vue:34]: Empty styles

File: src/store/modules/issues.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          commit('mutateList', result.data)
+        })
+        .catch(e => {
+          commit('mutateList', [])
+        })
+        .finally(() => commit('mutateLoading', false))
+    },
+    async fetchOne ({ commit }, params) {
+      commit('mutateLoading', true)
+      let result = await fetchIssue(params)
```
Comment (rapi) [inline src/store/modules/issues.js:25]: All requests shold be under try/catch

File: src/store/modules/issues.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    },
+    mutateLoading: (state, loading) => {
+      state.loading = loading
+    }
+  },
+  getters: {
+    getById: (state) => (id) => state.list.find(element => element.number === id),
+    getList: (state) => state.list,
+    getListPaginated: (state) => (params) => {
+      return state.list.slice(params.top * (params.page - 1), params.top * params.page)
```
Comment (rapi) [inline src/store/modules/issues.js:42]: Page should be taken from backend

File: src/components/issues/List.vue [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+  watch: {
+    pageNumber: function (val, oldVal) {
+      this.$router.push({ name: 'home', params: { page: val } })
+    }
+  },
+  computed: {
+    ...mapGetters({
+      issues: 'issues/getList',
+      loading: 'issues/getLoading',
+      getListPaginated: 'issues/getListPaginated'
```
Comment (rapi) [inline src/components/issues/List.vue:79]: Make request to the backend with current page

**Review by rapi — COMMENTED** [review]
Logic with pagination
Eslint format
Empty styles

File: src/plugins/axios.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios'
+
+const instance = axios.create({
```
Comment (gregurco) [inline src/plugins/axios.js:3]: you already have instance of axios. This is useless.

**Pull request merged**